### PR TITLE
Test reconnection when C* unavailable at startup

### PR DESCRIPTION
**Pull request by chbatey**

### CODE REVIEW

File: core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+object ReconnectSpec {
+  val freePort = SocketUtil.temporaryLocalPort()
+  val config = ConfigFactory.parseString(s"""
+      datastax-java-driver {
+        basic.load-balancing-policy.local-datacenter = "datacenter1"
+        // Will fail without this setting 
+        advanced.reconnect-on-init = true
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala]: right, that's documented as an important property.
could you also add a comment about it in our reference.conf.

@ennru this property should be recommended in Alpakka docs also, if it's not already

File: core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    with ImplicitSender
+    with AnyWordSpecLike
+    with Matchers
+    with ScalaFutures {
+
+  "Reconnecting" must {
+    "start with system off" in {
+      val pa = system.actorOf(Props(new AwaitPersistenceInit("pid", "", "")))
+      pa ! "hello"
+      expectNoMessage()
```
Comment (patriknw) [inline core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala]: it should also terminate? or does that take too long time?

File: core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala (hunk 1 above)
Comment (ennru) [reply core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala]: Thanks, I added https://github.com/akka/alpakka/issues/2218

File: core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala (hunk 2 above)
Comment (chbatey) [reply core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala]: Can add terminate check

File: core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala (hunk 1 above)
Comment (chbatey) [reply core/src/test/scala/akka/persistence/cassandra/ReconnectSpec.scala]: Yes i'll add it to reference

**Comment by chbatey** [comment]
Added note in reference.conf, termination takes a long time so left it out of the spec

**Pull request merged**